File: backend/job-application-service/src/main/java/com/jobportal/jobapp/entities/JobApplication.java
package com.jobportal.jobapp.entities;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "job_application")
public class JobApplication {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "application_id")
    private Integer applicationId;

    @Column(name = "job_id")
    private Integer jobId;

    @Column(name = "cid")
    private Integer candidateId;

    @Column(name = "resume_id")
    private Integer resumeId;

    @Column(name = "application_date")
    private LocalDateTime applicationDate = LocalDateTime.now();

    @ManyToOne
    @JoinColumn(name = "status")
    private JobStatus status;

    public JobApplication() {
    }

    public Integer getApplicationId() {
        return applicationId;
    }

    public void setApplicationId(Integer applicationId) {
        this.applicationId = applicationId;
    }

    public Integer getJobId() {
        return jobId;
    }

    public void setJobId(Integer jobId) {
        this.jobId = jobId;
    }

    public Integer getCandidateId() {
        return candidateId;
    }

    public void setCandidateId(Integer candidateId) {
        this.candidateId = candidateId;
    }

    public Integer getResumeId() {
        return resumeId;
    }

    public void setResumeId(Integer resumeId) {
        this.resumeId = resumeId;
    }

    public LocalDateTime getApplicationDate() {
        return applicationDate;
    }

    public void setApplicationDate(LocalDateTime applicationDate) {
        this.applicationDate = applicationDate;
    }

    public JobStatus getStatus() {
        return status;
    }

    public void setStatus(JobStatus status) {
        this.status = status;
    }
}

File: backend-dotnet/Services/JobPortal.JobApplicationService/Entities/JobApplication.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JobPortal.JobApplications.Entities;

[Table("job_application")]
public class JobApplication
{
    [Key]
    [Column("application_id")]
    public int ApplicationId { get; set; }

    [Column("job_id")]
    public int? JobId { get; set; }

    [Column("cid")]
    public int? CandidateId { get; set; }

    [Column("resume_id")]
    public int? ResumeId { get; set; }

    [Column("application_date")]
    public DateTime ApplicationDate { get; set; } = DateTime.Now;

    [Column("status")]
    public int? StatusId { get; set; }

    [ForeignKey("StatusId")]
    public JobStatus? Status { get; set; }
}
